services/aggregation-svc/internal/tasks/tasks.go

```diff
@@ -0,0 +1,67 @@
+package tasks
+
+import (
+	"common"
+	"context"
+	pb "gen/proto/services/aggregation_svc/v1"
+	task_svc_pb "gen/proto/services/task_svc/v1"
+	daprc "github.com/dapr/go-sdk/client"
+	zlog "github.com/rs/zerolog/log"
+	"hwutil"
+)
+
+type ServiceServer struct {
+	pb.UnimplementedTasksServiceServer
+	daprClient *daprc.GRPCClient
+	wardSvc    task_svc_pb.WardServiceClient
+	roomSvc    task_svc_pb.RoomServiceClient
+	bedSvc     task_svc_pb.BedServiceClient
+	taskSvc    task_svc_pb.TaskServiceClient
+}
+
+func NewServiceServer(daprClient *daprc.GRPCClient) *ServiceServer {
+	service := ServiceServer{}
+	service.wardSvc = task_svc_pb.NewWardServiceClient(daprClient.GrpcClientConn())
+	service.roomSvc = task_svc_pb.NewRoomServiceClient(daprClient.GrpcClientConn())
+	service.bedSvc = task_svc_pb.NewBedServiceClient(daprClient.GrpcClientConn())
+	service.taskSvc = task_svc_pb.NewTaskServiceClient(daprClient.GrpcClientConn())
+	return &service
+}
+
+func (s ServiceServer) GetWardsOverview(ctx context.Context, _ *pb.GetWardsOverviewRequest) (*pb.GetWardsOverviewResponse, error) {
+	log := zlog.Ctx(ctx)
+
+	ctx, cancel := common.PrepCtxForSvcToSvcCall(ctx, "task-svc")
+	defer cancel()
+
+	res, err := s.wardSvc.GetWards(ctx, &task_svc_pb.GetWardsRequest{})
+	if err != nil {
+		return nil, err
+	}
+
+	wards := hwutil.Map(res.Wards, func(ward *task_svc_pb.GetWardsResponse_Ward) *pb.GetWardsOverviewResponse_Ward {
+		res, err := s.roomSvc.GetRoomsByWard(ctx, &task_svc_pb.GetRoomsByWardRequest{WardId: ward.Id})
+		if err != nil {
+			log.Error().Err(err).Send()
+			return nil
+		}
+
+		var bedCount uint32
+		for _, room := range res.Rooms {
+			res, err := s.bedSvc.GetBedsByRoom(ctx, &task_svc_pb.GetBedsByRoomRequest{RoomId: room.Id})
+			if err != nil {
+				log.Error().Err(err).Send()
+				return nil
+			}
+			bedCount += uint32(len(res.Beds))
+		}
+
+		return &pb.GetWardsOverviewResponse_Ward{
+			Id:       ward.Id,
+			Name:     ward.Name,
+			BedCount: bedCount,
+		}
+	})
+
+	return &pb.GetWardsOverviewResponse{Wards: wards}, nil
+}
```

Comment:
I would move that to the ward service. For me, the entire aggregation service is unnecessary.

Reply:
As already discussed ...
This pr will be migrated to `feat: ward-svc` after the next push